lib/dor/workflow/client/lifecycle_routes.rb

```diff
@@ -10,33 +11,106 @@ def initialize(requestor:)
         end
 
         # Returns the Date for a requested milestone from workflow lifecycle
-        # @param [String] repo repository name
+        #
+        # @param [String] repo The repository the object resides in. This parameter is deprecated
         # @param [String] druid object id
         # @param [String] milestone_name the name of the milestone being queried for
         # @param [Number] version the version to query for
         # @param [Boolean] active_only (false) if true, return only lifecycle steps for versions that have all processes complete
         # @return [Time] when the milestone was achieved.  Returns nil if the milestone does not exist
         #
-        def lifecycle(repo, druid, milestone_name, version: nil, active_only: false)
-          filter_milestone(query_lifecycle(repo, druid, version: version, active_only: active_only), milestone_name)
+        def lifecycle(*args)
+          case args.size
+          when 5
+            Deprecation.warn(self, 'you provided 5 args, but lifecycle now takes kwargs')
```

Comment:
Why are  we supporting 5 args?  We've never done that before, right?

Reply:
the existing (as of today) args are: repo, druid, milestone_name, version: nil, active_only: false

Reply:
oh, are version and active_only actually the 4th arg together? That would make sense.